proseco/catalog.py

```diff
@@ -0,0 +1,94 @@
+import traceback
+import numpy as np
+
+from .core import ACACatalogTable
+from .guide import get_guide_catalog, GuideTable
+from .acq import get_acq_catalog, AcqTable
+from .fid import get_fid_catalog, FidTable
+
+
+def get_aca_catalog(**kwargs):
+    try:
+        aca = _get_aca_catalog(**kwargs)
+    except Exception:
+        #aca = ACACatalogTable.empty()  # TBD Makes zero-length table with correct columns
+        aca = ACACatalogTable()
+        aca.meta['exception'] = traceback.format_exc()
+
+        if aca.acqs is None:
+            #aca.acqs = AcqTable.empty()
+            aca.acqs = []
+        if aca.fids is None:
+            #aca.fids = FidTable.empty()
+            aca.fids = []
+        if aca.guides is None:
+            #aca.guides = GuideTable.empty()
+            aca.guides = []
+    return aca
+
+def _get_aca_catalog(obsid=0, att=None, man_angle=None, date=None,
+                     dither_acq=None, dither_guide=None,
+                     t_ccd_acq=None, t_ccd_guide=None,
+                     detector=None, sim_offset=None, focus_offset=None,
+                     n_guide=None, n_fid=None, n_acq=None, monitor_windows=None,
+                     include_ids=None, include_halfws=None,
+                     print_log=False):
+
+    catalog = ACACatalogTable(print_log=print_log)
+    # Put at least the obsid in top level meta for now
+    catalog.meta['obsid'] = obsid
+
+    catalog.acqs = get_acq_catalog(obsid=obsid, att=att, date=date, t_ccd=t_ccd_acq,
+                                   dither=np.max(dither_acq), man_angle=man_angle,
+                                   print_log=print_log)
+    catalog.fids = get_fid_catalog(detector=detector, sim_offset=sim_offset,
+                                   focus_offset=focus_offset, acqs=catalog.acqs,
+                                   stars=catalog.acqs.meta['stars'], print_log=print_log)
+    catalog.guides = get_guide_catalog(obsid=obsid, att=att, date=date, t_ccd=t_ccd_guide,
+                                       n_guide=n_guide, dither=dither_guide,
+                                       stars=catalog.acqs.meta['stars'], print_log=print_log)
+
+    merge_cat = merge_cats(fids=catalog.fids, guides=catalog.guides, acqs=catalog.acqs,
+                           mons=monitor_windows)
+    for col in merge_cat.colnames:
+        catalog[col] = merge_cat[col]
+
+    return catalog
+
+
+def merge_cats(fids=[], guides=[], acqs=[], mons=[]):
```

Comment:
Should never use `[]` in a function def, this is a common gateway for difficult to understand bugs.  Use `None` instead.